### ISSUE TITLE
com_plugins modal not handling close correct

### ISSUE THREAD

**Issue by Ruud68** [joomla/joomla-cms#18285]
### Steps to reproduce the issue
1. open com_redirect
2. in notice link click on 'Redirect System Plugin'
... modal with redirect plugin is opened
3. click on the close button (or on the close X) in the modal

### Expected result
plg_system_redirect should be checked in again

### Actual result
plg_system_redirect is not checked in, but still checked out.

### System information (as much as possible)
PHP Built On 	Linux ruud-development 4.10.0-35-generic #39-Ubuntu SMP Wed Sep 13 07:46:59 UTC 2017 x86_64
Database Version 	5.7.19-0ubuntu0.17.04.1
Database Collation 	utf8_general_ci
Database Connection Collation 	utf8mb4_general_ci
PHP Version 	7.0.24-1+ubuntu17.04.1+deb.sury.org+1
Web Server 	Apache/2.4.25 (Ubuntu)
WebServer to PHP Interface 	apache2handler
Joomla! Version 	Joomla! 3.8.1 Stable [ Amani ] 4-October-2017 14:00 GMT
Joomla! Platform Version 	Joomla Platform 13.1.0 Stable [ Curiosity ] 24-Apr-2013 00:00 GMT
User Agent 	Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:56.0) Gecko/20100101 Firefox/56.0

### Additional comments
the close button can be 'fixed' by replacing the following code in ./administrator/components/com_redirect/views/links/tmpl/default.php
find
```
'<button class="btn" data-dismiss="modal" aria-hidden="true">'
						. JText::_("JLIB_HTML_BEHAVIOR_CLOSE") . '</button>'
```

replace with
```
'<button type="button" class="btn" data-dismiss="modal" aria-hidden="true"'
							. ' onclick="jQuery(\'#plugin' . $this->redirectPluginId . 'Modal iframe\').contents().find(\'#closeBtn\').click();">'
							. JText::_('JLIB_HTML_BEHAVIOR_CLOSE') . '</button>'
```

The behavior of the X button just bypasses all close logic for the plugin and 'hard' closes the modal.
Code for this is (or should be?) in ./administrator/components/com_plugins/views/plugin/tmpl/modal.php

I have unfortunately no clue how to fix that :(

**Comment by brianteeman** [comment]
at a guess this would be the same for all modals

**Comment by Ruud68** [comment]
just tested with articles modal (via Single Article menu > edit).
the modal does NOT have the X close on top (so that issue is 'solved', and the close button is correctly doing the article check in<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18285">issues.joomla.org/tracker/joomla-cms/18285</a>.</sub>

**Comment by Ruud68** [comment]
maiden change :) https://github.com/joomla/joomla-cms/pull/18292<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18285">issues.joomla.org/tracker/joomla-cms/18285</a>.</sub>

**Comment by ghost** [comment]
closed as having Pull Request #18292
